### PR TITLE
Fixed #32962 -- Consolidated string concatenations in BaseDatabaseSchemaEditor.column_sql().

### PR DESCRIPTION
**Pull request by cjerdonek**
https://code.djangoproject.com/ticket/32962

### CODE REVIEW

File: django/db/backends/base/schema.py [hunk 1]
```diff
@@ -241,36 +231,51 @@ def column_sql(self, model, field, include_default=False):
...
+            yield str(self.connection.ops.tablespace_sql(tablespace, inline=True))
+
+    def column_sql(self, model, field, include_default=False):
+        """
+        Take a field and return its column definition.
+        The field must already have had set_attributes_from_name() called.
+        """
+        # Get the column's type and use that as the basis of the SQL
+        db_params = field.db_parameters(connection=self.connection)
+        sql_type = db_params['type']
```
Comment (felixxm) [inline django/db/backends/base/schema.py:270]: Maybe:
```suggestion
        column_db_type = db_params['type']
```

**Review by felixxm — COMMENTED** [review]
@cjerdonek Thanks :+1:

**Comment by cjerdonek** [comment]
Thanks, @felixxm! I also went ahead and cleaned the comments and docstring in the method.

File: django/db/backends/base/schema.py [hunk 2]
```diff
@@ -241,36 +230,50 @@ def column_sql(self, model, field, include_default=False):
         )
         if include_default:
             default_value = self.effective_default(field)
-            column_default = ' DEFAULT ' + self._column_default_sql(field)
             if default_value is not None:
+                column_default = 'DEFAULT ' + self._column_default_sql(field)
```
Comment (cjerdonek) [inline django/db/backends/base/schema.py:234]: I noticed this could be moved one line down, FYI.

File: django/db/backends/base/schema.py [hunk 3]
```diff
@@ -241,36 +230,50 @@ def column_sql(self, model, field, include_default=False):
...
             null = True
-        if null and not self.connection.features.implied_column_null:
-            sql += " NULL"
-        elif not null:
-            sql += " NOT NULL"
-        # Primary key/unique outputs
+        if not null:
+            yield 'NOT NULL'
+        elif not self.connection.features.implied_column_null:
+            yield 'NULL'
```
Comment (cjerdonek) [inline django/db/backends/base/schema.py:251]: I also noticed this simplification (added as a second commit).

**Comment by felixxm** [comment]
@cjerdonek Thanks :+1: Looks great :1st_place_medal:

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Pull request merged**